### PR TITLE
Add changelog entry for ``dask-expr`` switch

### PR DESCRIPTION
**Pull request by phofl**

### CODE REVIEW

**Comment by phofl** [comment]
cc @mrocklin if you have thoughts about the wording

File: docs/source/changelog.rst [hunk 1]
```diff
@@ -11,6 +11,39 @@ Released on December 15, 2023
...
+Logical Query Planning now available for Dask DataFrames
+""""""""""""""""""""""""""""""""""""""""""""""""""""""""
+
+Dask DataFrames now make a logical query planning layer available that makes queries
+more efficient. This feature is currently optionally available and requires an explicit opt
+in by users. Set
+
+```
+dask.config.set({"dataframe.query-planning": True})
+```
```
Comment (jrbourbeau) [inline docs/source/changelog.rst:23]: We should switch to sphinx syntax for this code block

File: docs/source/changelog.rst [hunk 2]
```diff
@@ -11,6 +11,39 @@ Released on December 15, 2023
...
+
+to activate it. Additionally, you have to install ``dask-expr`` with
+
+```
+pip install dask-expr
+```
```
Comment (jrbourbeau) [inline docs/source/changelog.rst:29]: And this one too

File: docs/source/changelog.rst [hunk 3]
```diff
@@ -11,6 +11,39 @@ Released on December 15, 2023
 Highlights
 ^^^^^^^^^^
 
+Logical Query Planning now available for Dask DataFrames
+""""""""""""""""""""""""""""""""""""""""""""""""""""""""
+
+Dask DataFrames now make a logical query planning layer available that makes queries
+more efficient. This feature is currently optionally available and requires an explicit opt
+in by users. Set
```
Comment (jrbourbeau) [inline docs/source/changelog.rst:19]:
```suggestion
more efficient. This feature is currently off by default, but can be turned on with:
```

File: docs/source/changelog.rst [hunk 4]
```diff
@@ -11,6 +11,39 @@ Released on December 15, 2023
...
+
+```
+dask.config.set({"dataframe.query-planning": True})
+```
+
+to activate it. Additionally, you have to install ``dask-expr`` with
```
Comment (jrbourbeau) [inline docs/source/changelog.rst:25]:
```suggestion
You also need to have ``dask-expr`` installed:
```

File: docs/source/changelog.rst [hunk 5]
```diff
@@ -11,6 +11,39 @@ Released on December 15, 2023
...
+
+```
+pip install dask-expr
+```
+
+to get access to the implementation. This feature is still under active development
```
Comment (jrbourbeau) [inline docs/source/changelog.rst:31]:
```suggestion
This feature is still under active development
```

File: docs/source/changelog.rst [hunk 6]
```diff
@@ -11,6 +11,39 @@ Released on December 15, 2023
...
+and the API isn't stable yet, so breaking changes can occur. We have seen promising
+performance improvements so far, see
+`this blog post <https://blog.coiled.io/blog/dask-expr-tpch-dask.html>`_
+and `our benchmarks <https://tpch.coiled.io>`_  for more information. The benchmarks
+are updated regularly as we enable more improvements.
+
+A more detailed explanation of how the query optimizer works can be found in
+`our introduction post <https://blog.coiled.io/blog/dask-expr-introduction.html>`_
```
Comment (jrbourbeau) [inline docs/source/changelog.rst:39]:
```suggestion
`this blogpost <https://blog.coiled.io/blog/dask-expr-introduction.html>`_.
```

File: docs/source/changelog.rst [hunk 7]
```diff
@@ -11,6 +11,39 @@ Released on December 15, 2023
...
+```
+
+to activate it. Additionally, you have to install ``dask-expr`` with
+
+```
+pip install dask-expr
+```
+
+to get access to the implementation. This feature is still under active development
+and the API isn't stable yet, so breaking changes can occur. We have seen promising
```
Comment (jrbourbeau) [inline docs/source/changelog.rst:32]:
```suggestion
and the `API <https://github.com/dask-contrib/dask-expr#api-coverage>`_ isn't stable yet, so breaking changes can occur. We have seen promising
```

File: docs/source/changelog.rst [hunk 8]
```diff
@@ -11,6 +11,39 @@ Released on December 15, 2023
...
+and `our benchmarks <https://tpch.coiled.io>`_  for more information. The benchmarks
+are updated regularly as we enable more improvements.
+
+A more detailed explanation of how the query optimizer works can be found in
+`our introduction post <https://blog.coiled.io/blog/dask-expr-introduction.html>`_
+
+The current API coverage can be found
+`here <https://github.com/dask-contrib/dask-expr#api-coverage>`_. We will add more
+documentation over the next few weeks.
+
```
Comment (jrbourbeau) [inline docs/source/changelog.rst:44]: Removing in favor of a link above

```suggestion
```

File: docs/source/changelog.rst [hunk 9]
```diff
@@ -11,6 +11,39 @@ Released on December 15, 2023
...
+We expect to make the query optimizer the default early next year.
+
```
Comment (jrbourbeau) [inline docs/source/changelog.rst:46]: Can you link to the relevant PR(s) / authors like is done in other changelog entries (for example here https://docs.dask.org/en/stable/changelog.html#pipinstall-restart-and-environment-variables)

File: docs/source/changelog.rst [hunk 10]
```diff
@@ -11,6 +11,39 @@ Released on December 15, 2023
 Highlights
 ^^^^^^^^^^
 
+Logical Query Planning now available for Dask DataFrames
+""""""""""""""""""""""""""""""""""""""""""""""""""""""""
+
+Dask DataFrames now make a logical query planning layer available that makes queries
+more efficient. This feature is currently optionally available and requires an explicit opt
```
Comment (jrbourbeau) [inline docs/source/changelog.rst:18]:
```suggestion
Dask DataFrames are now much more performant by using a logical query planner.
This feature is currently optionally available and requires an explicit opt
```

File: docs/source/changelog.rst [hunk 11]
```diff
@@ -11,6 +11,39 @@ Released on December 15, 2023
...
+
+```
+pip install dask-expr
+```
+
+to get access to the implementation. This feature is still under active development
+and the API isn't stable yet, so breaking changes can occur. We have seen promising
+performance improvements so far, see
+`this blog post <https://blog.coiled.io/blog/dask-expr-tpch-dask.html>`_
+and `our benchmarks <https://tpch.coiled.io>`_  for more information. The benchmarks
```
Comment (jrbourbeau) [inline docs/source/changelog.rst:35]:
```suggestion
and `these benchmarks <https://tpch.coiled.io>`_  for more information. The benchmarks
```

File: docs/source/changelog.rst [hunk 12]
```diff
@@ -11,6 +11,39 @@ Released on December 15, 2023
...
+are updated regularly as we enable more improvements.
```
Comment (jrbourbeau) [inline docs/source/changelog.rst:36]:
```suggestion
are updated regularly as improvements are made.
```

File: docs/source/changelog.rst (hunk 5 above)
Comment (jrbourbeau) [inline docs/source/changelog.rst:31]: Not a bid deal, but I'd switch the ordering here to say "We've seen promising performance improvements... The features is still under active development..."

**Review by jrbourbeau — COMMENTED** [review]
Thanks for taking the time to add this @phofl.

File: docs/source/changelog.rst (hunk 6 above)
Comment (phofl) [reply docs/source/changelog.rst:38]: sphinx will complain about using this twice, that's why I switched

File: docs/source/changelog.rst [hunk 13]
```diff
@@ -11,6 +11,39 @@ Released on December 15, 2023
 Highlights
 ^^^^^^^^^^
 
+Logical Query Planning now available for Dask DataFrames
```
[the remark this answers is not in the document]
Comment (phofl) [reply docs/source/changelog.rst:14]: Hm I'd like to keep the now available

File: docs/source/changelog.rst (hunk 1 above)
Comment (phofl) [reply docs/source/changelog.rst]: done

File: docs/source/changelog.rst (hunk 2 above)
Comment (phofl) [reply docs/source/changelog.rst]: same

File: docs/source/changelog.rst (hunk 5 above)
Comment (phofl) [reply docs/source/changelog.rst]: sure

File: docs/source/changelog.rst (hunk 6 above)
Comment (jrbourbeau) [reply docs/source/changelog.rst]: Ah, I see. You can use a trailing double underscore in that case to make the hyperlink anonymous (sphinx won't complain in that case)

```
`some text <https://some.link>`__
```

I just pushed a small commit that does this and a couple of other minor edits (hope that's okay)

**Review by jrbourbeau — APPROVED** [review]

**Pull request merged**